cogs/welcome/welcome.py

```diff
@@ -271,44 +303,45 @@ async def greetRemove(self, ctx, name):
         def check(message: discord.Message):
             return message.author == ctx.message.author and message.channel == ctx.message.channel
 
-        greetings = await self.config.guild(ctx.guild).KEY_GREETINGS()
-
-        tuple = ()
-        for greetingTuple in greetings:
-            if greetingTuple[0] == name:
-                await ctx.send(
-                    warning(
-                        "Are you sure you wish to delete this greeting? Respond with 'yes' if yes"
-                    )
+        greetings = await self.config.guild(ctx.guild).greetings()
+        if name in greetings:
+            await ctx.send(
+                warning(
+                    "Are you sure you wish to delete this greeting? Respond with 'yes' if yes"
                 )
-                tuple = greetingTuple
-                try:
-                    response = await self.bot.wait_for("message", timeout=30.0, check=check)
-                except asyncio.TimeoutError:
-                    await ctx.send("You took too long, not deleting")
-                    return
-
-                if response.content.lower() != "yes":
-                    await ctx.send("Not deleting")
-                    return
-        # delete the greeting
-        greetings.remove(tuple)
-        await ctx.send(f"{name} removed from list")
-        await self.config.guild(ctx.guild).KEY_GREETINGS.set(greetings)
+            )
+            try:
+                response = await self.bot.wait_for("message", timeout=30.0, check=check)
+            except asyncio.TimeoutError:
+                await ctx.send("You took too long, not deleting")
+                return
+
+            if response.content.lower() != "yes":
+                await ctx.send("Not deleting")
+                return
+
+        if name in greetings:
```

Comment:
we're already in the `if name in greetings` block, safe to remove
```suggestion
```

Reply:
done